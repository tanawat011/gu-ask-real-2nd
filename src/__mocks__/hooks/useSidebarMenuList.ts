export const useSidebarMenuList = () => {
  return [
    [
      {
        title: 'Dashboard',
        icon: 'dashboard',
        path: '/dashboard',
      },
      {
        title: 'Users',
        icon: 'user',
        path: '/users',
      },
      {
        title: 'Posts',
        icon: 'file',
        path: '/posts',
      },
    ],
    () => void 0,
  ]
}
