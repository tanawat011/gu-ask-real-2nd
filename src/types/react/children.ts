export type Children = { children: JSX.Element }
