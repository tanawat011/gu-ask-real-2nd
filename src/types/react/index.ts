export * from './children'
export * from './reactRouterDom'
