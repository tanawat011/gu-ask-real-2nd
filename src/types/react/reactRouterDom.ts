import type { Scroll } from 'types/common'

export type OutletContextProps = {
  scroll: Scroll
}
