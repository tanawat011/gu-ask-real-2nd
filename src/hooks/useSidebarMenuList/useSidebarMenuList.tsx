import type { MenuItem } from 'components/Sidebar'

import type { Dispatch, SetStateAction } from 'react'
import { useState } from 'react'

import { useRouteList } from 'hooks/useRouteList'
import { menuGenerator } from 'utils/route'

export const useSidebarMenuList = (): [
  MenuItem[],
  Dispatch<SetStateAction<MenuItem[]>>,
] => {
  const { main } = useRouteList()

  const [menuList, setMenuList] = useState<MenuItem[]>(menuGenerator(main._))

  return [menuList, setMenuList]
}
