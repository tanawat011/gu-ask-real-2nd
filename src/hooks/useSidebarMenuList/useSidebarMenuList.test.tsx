import { render } from '@testing-library/react'

import { useSidebarMenuList } from './useSidebarMenuList'

jest.mock('@fortawesome/react-fontawesome')

const ExampleComponent = () => {
  const [menuList] = useSidebarMenuList()

  return <div>{menuList[0].title}</div>
}

describe('useSidebarMenuList', () => {
  test('should return all menu list', () => {
    expect.hasAssertions()

    const { asFragment } = render(<ExampleComponent />)

    expect(asFragment()).toMatchSnapshot()
  })
})
