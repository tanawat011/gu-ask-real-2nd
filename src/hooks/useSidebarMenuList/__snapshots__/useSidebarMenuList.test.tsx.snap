// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useSidebarMenuList should return all menu list 1`] = `
<DocumentFragment>
  <div>
    APPS
  </div>
</DocumentFragment>
`;
